var/spack/repos/builtin/packages/exago/package.py

```diff
@@ -62,6 +66,24 @@ class Exago(CMakePackage, CudaPackage, ROCmPackage):
 
     depends_on("cmake@3.18:", type="build")
 
+    # Profiling and optimizations
+    depends_on("hiop+deepchecking~full_optimizations", when="+hiop~full_optimizations")
+    depends_on("hiop~deepchecking+full_optimizations", when="+hiop+full_optimizations")
+
+    # Force RelWithDebInfo when not using optimizations 
+    conflicts("build_type=Release", when="~full_optimizations", msg="Use RelWithDebInfo when not using optimizations")
+    conflicts("build_type=Debug", when="~full_optimizations", msg="Use RelWithDebInfo when not using optimizations")
+    conflicts("build_type=MinSizeRel", when="~full_optimizations", msg="Use RelWithDebInfo when not using optimizations")
```

Comment:
I wonder if there is a cleaner way of doing this. I know we struggled to configure a `build_type` initially, but perhaps we can clean this up in review.